plugins/incremental-ingestion-backend/src/database/IncrementalIngestionDatabaseManager.ts

```diff
@@ -0,0 +1,592 @@
+              .whereIn(
+                'ingestion_id',
+                ingestionIDs.map(entry => entry.id),
+              )
+          : [];
+
+      const markEntityIDs: { id: string }[] =
+        markIDs.length > 0
+          ? await tx('ingestion.ingestion_mark_entities')
+              .select('id')
+              .whereIn(
+                'ingestion_mark_id',
+                markIDs.map(entry => entry.id),
+              )
+          : [];
+
+      const markEntitiesDeleted = await this.deleteMarkEntities(
+        tx,
+        markEntityIDs,
+      );
+
+      const marksDeleted =
+        markIDs.length > 0
+          ? await tx('ingestion.ingestion_marks')
+              .delete()
+              .whereIn(
+                'ingestion_id',
+                ingestionIDs.map(entry => entry.id),
+              )
+          : 0;
+
+      const ingestionsDeleted = await tx('ingestion.ingestions')
+        .delete()
+        .where('provider_name', provider);
+
+      const next_action_at = new Date();
+      next_action_at.setTime(next_action_at.getTime() + 24 * 60 * 60 * 1000);
+
+      await this.insertIngestionRecord({
+        id: v4(),
+        next_action: 'rest',
+        provider_name: provider,
+        next_action_at,
+        ingestion_completed_at: new Date(),
+        status: 'resting',
+        completion_ticket: v4(),
+      });
+
+      return { provider, ingestionsDeleted, marksDeleted, markEntitiesDeleted };
+    });
+  }
+
+  /**
+   * Creates a new ingestion record.
+   * @param provider - string
+   * @returns A new ingestion record
+   */
+  async createProviderIngestionRecord(provider: string) {
+    const ingestionId = v4();
+    const nextAction = 'ingest';
+    try {
+      await this.insertIngestionRecord({
+        id: ingestionId,
+        next_action: nextAction,
+        provider_name: provider,
+        status: 'bursting',
+        completion_ticket: 'open',
+      });
+      return { ingestionId, nextAction, attempts: 0, nextActionAt: Date.now() };
+    } catch (_e) {
+      // Creating the ingestion record failed. Return undefined.
+      return undefined;
+    }
+  }
+
+  /**
+   * Computes which entities to remove, if any, at the end of a burst.
+   * @param provider - string
+   * @param ingestionId - string
+   * @returns All entities to remove for this burst.
+   */
+  async computeRemoved(provider: string, ingestionId: string) {
+    return await this.client.transaction(async tx => {
+      const removed: { entity: string; ref: string }[] = await tx(
+        'final_entities',
+      )
+        .select(
+          tx.ref('final_entity').as('entity'),
+          tx.ref('refresh_state.entity_ref').as('ref'),
+        )
+        .join(
+          'refresh_state',
+          'refresh_state.entity_id',
+          'final_entities.entity_id',
+        )
+        .whereRaw(
+          `((final_entity::json #>> '{metadata, annotations, ${INCREMENTAL_ENTITY_PROVIDER_ANNOTATION}}')) = ?`,
```

Comment:
Not using the index here? Anyway, could just merge with the search table as mentioned

Reply:
Done!